lib/src/controls/yaru_popup_menu_button.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:flutter/material.dart';
+import 'package:yaru_icons/yaru_icons.dart';
+import '../constants.dart';
+
+/// A generic wrapper around [PopupMenuButton] that is visually more consistent
+/// to buttons and dialogs than [DropdownButton]
+class YaruPopupMenuButton<T> extends StatelessWidget {
+  const YaruPopupMenuButton({
+    super.key,
+    required this.initialValue,
+    required this.child,
+    required this.items,
+    this.onSelected,
+    this.onCanceled,
+    this.tooltip,
+    this.position = PopupMenuPosition.under,
+  });
+
+  final T initialValue;
+  final Widget child;
+  final List<PopupMenuItem<T>> items;
+  final Function(T)? onSelected;
+  final Function()? onCanceled;
+  final String? tooltip;
+  final PopupMenuPosition position;
+
+  @override
+  Widget build(BuildContext context) {
+    return DefaultTextStyle(
+      style: Theme.of(context).textTheme.bodyMedium ??
+          TextStyle(
+            fontSize: 20,
+            color: Theme.of(context).colorScheme.onSurface,
+          ),
+      child: ClipRRect(
+        borderRadius: BorderRadius.circular(kYaruButtonRadius),
+        child: Material(
+          color: Colors.transparent,
+          child: PopupMenuButton(
+            padding: EdgeInsets.zero,
+            initialValue: initialValue,
+            onSelected: onSelected,
+            onCanceled: onCanceled,
+            tooltip: tooltip,
+            itemBuilder: (context) {
+              return items;
```

Comment:
could we keep it as a builder? one can easily do it like this in an app, but it would be nice to have the opportunity to lazily build the items in their appropriate states at the time the button is pressed. if it's not a builder, you need to build the items in advance and rebuild the button whenever any popup item would change even when there's no popup, which is most of the time. :)